x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlResolveFieldsAction.java

```diff
@@ -6,59 +6,355 @@
-    protected void doExecute(Task task, FieldCapabilitiesRequest request, final ActionListener<FieldCapabilitiesResponse> listener) {
-        fieldCapsAction.executeRequest(task, request, this::executeLinkedRequest, listener);
+    protected void doExecute(Task task, FieldCapabilitiesRequest request, final ActionListener<EsqlResolveFieldsResponse> listener) {
+        executeRequest(task, request, listener);
     }
 
-    void executeLinkedRequest(
-        TransportService transportService,
-        Transport.Connection conn,
-        FieldCapabilitiesRequest request,
-        ActionListenerResponseHandler<FieldCapabilitiesResponse> responseHandler
+    public void executeRequest(Task task, FieldCapabilitiesRequest request, ActionListener<EsqlResolveFieldsResponse> listener) {
+        // workaround for https://github.com/elastic/elasticsearch/issues/97916 - TODO remove this when we can
+        searchCoordinationExecutor.execute(ActionRunnable.wrap(listener, l -> doExecuteForked(task, request, l)));
+    }
+
+    private void doExecuteForked(Task task, FieldCapabilitiesRequest request, ActionListener<EsqlResolveFieldsResponse> listener) {
+        if (request.isMergeResults()) {
+            throw new IllegalArgumentException("merging results not supported");
+        }
+        if (ccsCheckCompatibility) {
+            checkCCSVersionCompatibility(request);
+        }
+        final Executor singleThreadedExecutor = TransportFieldCapabilitiesAction.buildSingleThreadedExecutor(
+            searchCoordinationExecutor,
+            LOGGER
+        );
+        assert task instanceof CancellableTask;
+        final CancellableTask fieldCapTask = (CancellableTask) task;
+        // retrieve the initial timestamp in case the action is a cross cluster search
+        long nowInMillis = request.nowInMillis() == null ? System.currentTimeMillis() : request.nowInMillis();
+        ClusterState clusterState = clusterService.state();
+        ProjectState projectState = projectResolver.getProjectState(clusterState);
+        AtomicReference<TransportVersion> minTransportVersion = new AtomicReference<>(clusterState.getMinTransportVersion());
+        final Map<String, OriginalIndices> remoteClusterIndices = transportService.getRemoteClusterService()
+            .groupIndices(request.indicesOptions(), request.indices(), request.returnLocalAll());
+        final OriginalIndices localIndices = remoteClusterIndices.remove(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY);
+        final String[] concreteIndices;
+        if (localIndices == null) {
+            // in the case we have one or more remote indices but no local we don't expand to all local indices and just do remote indices
+            concreteIndices = Strings.EMPTY_ARRAY;
+        } else {
+            concreteIndices = indexNameExpressionResolver.concreteIndexNames(projectState.metadata(), localIndices);
+        }
+
+        if (concreteIndices.length == 0 && remoteClusterIndices.isEmpty()) {
```

Comment:
We're not requesting the remote minTV if the remote has no indices? Do we assume that "no indices" means "no remote" or something like that?

Reply:
We shouldn't be sending to that remote at all in that case. Checking.